pinot-controller/src/main/java/org/apache/pinot/controller/validation/RealtimeSegmentValidationManager.java

```diff
@@ -67,6 +68,23 @@ public RealtimeSegmentValidationManager(ControllerConf config, PinotHelixResourc
     Preconditions.checkState(_segmentLevelValidationIntervalInSeconds > 0);
   }
 
+  // TODO: Fix the race condition when controller leadership may not be decided by the time the method is called
```

Comment:
Is there an issue created for this? If not, please create an issue, and add the issue number here. How and when do you plan to fix this race condition? It should be sooner than later. You may want to get another PR ready for that really quickly. 

If you think it is going to take a very long time, then we should just wait until we merge this PR. Otherwise, the subtlety of the race condition will be lost.